### PR TITLE
Add requirements.txt, update gh-page_builder.yml

### PR DESCRIPTION
**Pull request by halehawk**
Update github workflow action, this action only works after the user selects: settings->pages: Build and deployment: choose Github Actions from the dropdown menu.

### CODE REVIEW

File: .github/workflows/gh-page_builder.yml [hunk 1]
```diff
@@ -11,29 +16,35 @@ jobs:
...
-        sudo chown -R $(whoami):$(whoami) .
-        git config --global user.email "$GITHUB_ACTOR@users.noreply.github.com"
-        git config --global user.name "$GITHUB_ACTOR"
-        git remote set-url origin "https://$GITHUB_ACTOR:${{ secrets.GITHUB_TOKEN }}@github.com/$GITHUB_REPOSITORY"
-        ls -lrt
-        ghp-import _build/html -f -p -n
+    - name: Upload artifact
+      uses: actions/upload-pages-artifact@v3
+      with:
+        path: ./_build/html
```
Comment (negin513) [inline .github/workflows/gh-page_builder.yml:39]: @halehawk do we need this step?

File: README.md [hunk 2]
```diff
@@ -1,6 +1,7 @@
 ![SEA 2024 Logo](assets/logo_3.png)
 
 ># SEA ISS 2024 Notebook Submission Template 📓
+[![JupyterBook](https://github.com/UCAR-SEA/SEA-ISS-Template/actions/workflows/gh-page_builder.yml/badge.svg)](https://github.com/UCAR-SEA/SEA-ISS-Template/actions/workflows/gh-page_builder.yml)
```
Comment (negin513) [inline README.md:4]: Thanks for adding this badge.

File: .github/workflows/gh-page_builder.yml [hunk 3]
```diff
@@ -3,6 +3,11 @@ name: JupyterBook
 
 on:
   [push]
+  
+permissions:
+  contents: read
+  pages: write
+  id-token: write
```
Comment (negin513) [inline .github/workflows/gh-page_builder.yml:10]: Do we need the `id-token: write` ? This gives Github Workflow the ability to create a token ID.

File: .github/workflows/gh-page_builder.yml (hunk 1 above)
Comment (halehawk) [reply .github/workflows/gh-page_builder.yml:39]: @negin513 Yes, this step can zip all the html files and upload to the website.

File: .github/workflows/gh-page_builder.yml (hunk 3 above)
Comment (halehawk) [reply .github/workflows/gh-page_builder.yml:10]: Yes, `pages: write` is to deploy to Pages, `id-token: write` is to verify the deployment originates from an appropriate source. These two permission grant GITHUB_TOKEN the permissions required to make a Pages deployment.

**Comment by negin513** [comment]
Hello @halehawk, Thanks so much for working on this. This looks good. 

I will merge in.

**Comment by negin513** [comment]
Hello @halehawk , After merging this, it seems like the workflow failed. I will take a closer look tonight and will revert back to not use `id-token`.

**Comment by halehawk** [comment]
It looks like the deploy failed when you merge my PR, but why does the
deploy have no any logs?

**Comment by halehawk** [comment]
I think you didn't turn on using the github actions to deploy. Could
you please check that you did the following?
Update github workflow action, this action only works after the user
selects: settings->pages: Build and deployment: choose Github Actions
from the dropdown menu.

**Pull request merged**